File: app/src/main/java/com/mohsenoid/rickandmorty/view/episode/list/EpisodeListContract.java
package com.mohsenoid.rickandmorty.view.episode.list;

import com.mohsenoid.rickandmorty.domain.entity.EpisodeEntity;
import com.mohsenoid.rickandmorty.view.base.BasePresenter;
import com.mohsenoid.rickandmorty.view.base.BaseView;

import java.util.List;

public interface EpisodeListContract {

    interface View extends BaseView {

        void showLoadingMore();

        void hideLoadingMore();

        void setEpisodes(List<EpisodeEntity> episodes);

        void updateEpisodes(List<EpisodeEntity> episodes);

        void reachedEndOfList();
    }

    interface Presenter extends BasePresenter<View> {

        void loadEpisodes();

        void loadMoreEpisodes(int page);
    }
}

File: app/src/main/kotlin/com/mohsenoid/rickandmorty/view/episode/list/EpisodeListContract.kt
package com.mohsenoid.rickandmorty.view.episode.list

import com.mohsenoid.rickandmorty.domain.model.ModelEpisode
import com.mohsenoid.rickandmorty.view.base.BasePresenter
import com.mohsenoid.rickandmorty.view.base.BaseView

interface EpisodeListContract {

    interface View : BaseView {

        fun showLoadingMore()

        fun hideLoadingMore()

        fun setEpisodes(episodes: List<ModelEpisode>)

        fun updateEpisodes(episodes: List<ModelEpisode>)

        fun reachedEndOfList()
    }

    abstract class Presenter : BasePresenter<View>() {

        abstract suspend fun loadEpisodes()

        abstract suspend fun loadMoreEpisodes(page: Int)
    }
}
